system_test.go

```diff
@@ -1,287 +1,106 @@
 package main
 
 import (
-	"bytes"
 	"encoding/json"
-	"fmt"
-	"io/ioutil"
 	"os"
-	"os/exec"
-	"sort"
 	"strings"
-	"syscall"
 	"testing"
 	"time"
 
 	"github.com/Shopify/sarama"
-	qt "github.com/frankban/quicktest"
+	"github.com/google/go-cmp/cmp"
+	"github.com/rogpeppe/go-internal/testscript"
 )
 
-type cmd struct {
-	in string
+func TestMain(m *testing.M) {
+	os.Exit(testscript.RunMain(m, map[string]func() int{
+		"kt": func() int {
+			main()
+			// TODO make main return exit code.
```

Comment:
Why not address this?  With that we can use these in CI.

Reply:
I'd like to address this, but it means a significant code refactor across all the subcommands because the code currently relies fairly heavily on calling os.Exit. I think that's better left for another PR, where I might actually change the whole way that subcommands are done.